app/models/manager_refresh/inventory_collection.rb

```diff
@@ -354,29 +322,155 @@ def inspect
       to_s
     end
 
-    def actualize_dependency(key, value)
-      if dependency?(value)
-        (dependency_attributes[key] ||= Set.new) << value.inventory_collection
-        transitive_dependency_attributes << key if transitive_dependency?(value)
-      elsif value.kind_of?(Array) && value.any? { |x| dependency?(x) }
-        (dependency_attributes[key] ||= Set.new) << value.detect { |x| dependency?(x) }.inventory_collection
-        transitive_dependency_attributes << key if value.any? { |x| transitive_dependency?(x) }
+    def scan!
+      data.each do |inventory_object|
+        scan_inventory_object(inventory_object)
       end
     end
 
+    def db_collection_for_comparison
+      return arel unless arel.nil?
+      parent.send(association)
+    end
+
     private
 
-    attr_writer :attributes_blacklist, :attributes_whitelist
+    attr_writer :attributes_blacklist, :attributes_whitelist, :db_data_index, :references
+
+    # Finds manager_uuid in the DB. Using a configured strategy we cache obtained data in the db_data_index, so the
+    # same find will not hit database twice. Also if we use lazy_links and this is called when
+    # data_collection_finalized?, we load all data from the DB, referenced by lazy_links, in one query.
+    #
+    # @param manager_uuid [String] a manager_uuid of the InventoryObject we search in the local DB
+    def find_in_db(manager_uuid)
+      # TODO(lsmola) selected need to contain also :keys used in other InventoryCollections pointing to this one, once
```

Comment:
right, this commented code is pretty old actually, I just moved it. It's for the next step when I will track what attributes are being referenced and I will select only those